firedrake/function.py

```diff
@@ -489,6 +489,8 @@ def at(self, arg, *args, **kwargs):
         """
         # Need to ensure data is up-to-date for reading
         self.dat._force_evaluation(read=True, write=False)
+        self.dat.global_to_local_begin(op2.READ)
+        self.dat.global_to_local_end(op2.READ)
```

Comment:
Yes, because the computation phase guarantees only that owned points are correct once the computation has finished.  We need this global to local phase to get up to date ghosts.